auth/data/src/main/di/AuthDataModule.kt

```diff
@@ -0,0 +1,30 @@
+package com.maasbodev.auth.data.di
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.ViewModelComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(ViewModelComponent::class)
+object AuthDataModule {
+
+	@Provides
+	fun provideAuthRepository(
+		httpClient: HttpClient,
+		sessionStorage: SessionStorage,
+	): AuthRepository {
+		return AuthRepositoryImpl(httpClient)
+	}
+
+	@Provides
+	fun provideHttpClient(): HttpClient {
+		return HttpClient()
+	}
+
+	@Provides
+	fun provideUserDataValidator(patternvalidator: PatternValidator): UserDataValidator {
+		return UserDataValidator()
+	}
+}
```

Comment:
This module is probably the reason why you get the errors. If you want to make dependencies available just for a ViewModel, also all the dependencies these rely on have to be scoped to that ViewModel. So if you have a ViewModel-scoped AuthRepositoryImpl, also the httpClient and sessionStorage need to be scoped to the ViewModel. For the session storage, that doesn't seem to be the case here since it's a singleton.

I would keep them here just singletons.

Reply:
The session storage is bonded in another model, I thought Hilt would find it

Reply:
It does find it, but it has a different scope. The session storage is installed in `SingletonComponent` while this one is installed in `ViewModelComponent`. The scopes have to match

Reply:
Both are @InstallIn(ViewModelComponent::class), is not this what gives the scope?